Core/Executor/UserManager.php

```diff
@@ -111,25 +115,29 @@ protected function update()
 
                 $assignedGroups = $userService->loadUserGroupsOfUser($user);
 
+                $targetGroupIds = [];
                 // Assigning new groups to the user
                 foreach ($this->dsl['groups'] as $groupToAssignId) {
+                    $groupId = $this->referenceResolver->resolveReference($groupToAssignId);
+                    $groupToAssign = $this->userGroupMatcher->matchOneByKey($groupId);
```

Comment:
Exactly, that's what I meant with "allowing usage of user group remote_id in user's "groups" when creating/updating"